## Charities in Colorado: Initial EDA and data preprocessing

### Data source: Registration data for charities - CO Secretary of State
https://www.sos.state.co.us/pubs/charities/CCSAreports.html?faces-redirect=true 
Carities (XLS) link under the Data Extacts


In [1]:
# Imports
import os
import pandas as pd
import numpy as np
import datetime
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
# load data
charities_all = pd.read_excel('CoCodeCo/datafiles/CharitableOrganizationsDataExtract.xlsx', parsedates=True)

In [3]:
#Check the data load
charities_all.head()

Principal Name of Organization  \
0   DENVER HARLEQUINS RUGBY FOOTBALL CLUB   
1                    #WALKAWAY FOUNDATION   
2                    1 LIQUID HOUSE, INC.   
3                     1/20/21 ACTION FUND   
4                 1% FOR THE PLANET, INC.   

                                                DBAs  Registration Number  \
0                                                NaN         2.018301e+10   
1                                                NaN         2.019301e+10   
2                                                NaN         2.003301e+10   
3                                                NaN         2.019300e+10   
4  1 FOR THE PLANET, 1% FOR THE PLANET, ONE PERCE...         2.016303e+10   

          EIN      Principal Address 1 Principal Address 2 Principal City  \
0  27-1686134          3843 XAVIER ST.                 NaN         DENVER   
1  83-2820906     441 NORTH LEE STREET           SUITE 100     ALEXANDRIA   
2  73-1267224           6668 LYNX COVE                 NaN      LITTLETON   
3  83-2210730       2370 MARKET STREET               # 433  SAN FRANCISCO   
4  91-2151932  47 MAPLE ST., SUITE 103                 NaN     BURLINGTON   

  Principal County Principal State Principal Zip         ...           \
0           DENVER              CO         80212         ...            
1  ALEXANDRIA CITY              VA         22314         ...            
2          DOUGLAS              CO         80124         ...            
3    SAN FRANCISCO              CA         94114         ...            
4       CHITTENDEN              VT          5401         ...            

  Fundraising Expenses Total Expenses  \
0                    0          26817   
1                    0         100000   
2                    0          16647   
3               275000        1993210   
4               170341        1469936   

  Fundraising Ratio Fundraising Expenses to Contributions  \
0                                             0.0000        
1                                             0.0000        
2                                             0.0000        
3                                             0.1310        
4                                             0.2865        

  Program Service Ratio Program Services Expenses Total Expenses  \
0                                             0.9836               
1                                             0.2000               
2                                             1.0000               
3                                             0.8319               
4                                             0.7454               

  Total Asset Amount Total Liabilities Amount Net Assets  \
0            4269.55                      0.0    4269.55   
1               0.00                      0.0       0.00   
2           16757.00                   3839.0   12918.00   
3          106790.00                      0.0  106790.00   
4         1044315.00                 238422.0  805893.00   

                                         NTEE Code 1  NTEE Code 2  \
0          S-COMMUNITY IMPROVEMENT,CAPACITY BUILDING  B-EDUCATION   
1                                        B-EDUCATION          NaN   
2                        A-ARTS,CULTURE & HUMANITIES          NaN   
3            R-CIVIL RIGHTS,SOCIAL ACTION,& ADVOCACY          NaN   
4  C-ENVIRONMENT QUALITY,PROTECTION & BEAUTIFICATION          NaN   

           NTEE Code 3  
0  N-RECREATION,SPORTS  
1                  NaN  
2                  NaN  
3                  NaN  
4                  NaN  

[5 rows x 36 columns]

In [4]:
#Check the data frame structure
charities_all.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17825 entries, 0 to 17824
Data columns (total 36 columns):
Principal Name of Organization                                    17824 non-null object
DBAs                                                              5839 non-null object
Registration Number                                               17824 non-null float64
EIN                                                               17824 non-null object
Principal Address 1                                               17825 non-null object
Principal Address 2                                               1980 non-null object
Principal City                                                    17824 non-null object
Principal County                                                  17549 non-null object
Principal State                                                   17818 non-null object
Principal Zip                                                     17824 non-null object
Phone       

In [5]:
charities_all.tail()

Principal Name of Organization  \
17820  ZOOLOGY FOUNDATION AT CROOKED WILLOW FARMS   
17821                       ZOOM TRACK CLUB, INC.   
17822     ZOOMERS, INC. DBA LAKE DILLON PRESCHOOL   
17823                         ZUMA'S RESCUE RANCH   
17824                                         NaN   

                                                    DBAs  Registration Number  \
17820                                                NaN         2.011304e+10   
17821                                                NaN         2.011302e+10   
17822  LAKE DILLON PRESCHOOL, LAKE DILLON PRESCHOOL A...         2.002301e+10   
17823  ALL SOULS RESCUE, ZUMA'S EXPERIENTIAL LEARNING...         2.012303e+10   
17824                                                NaN                  NaN   

              EIN                                Principal Address 1  \
17820  27-1125802                            10554 S. PERRY PARK RD.   
17821  68-0529237                                 6548 SERENGETI CIR   
17822  84-1139106                                  200 VILLAGE PLACE   
17823  80-0236203                                  7745 N. MOORE RD.   
17824         NaN  501 (3) (C) PUBLIC TRUST, 501 (C) (3), 501 (C)...   

      Principal Address 2 Principal City Principal County Principal State  \
17820                 NaN       LARKSPUR          DOUGLAS              CO   
17821                 NaN      LITTLETON          DOUGLAS              CO   
17822                 NaN         DILLON           SUMMIT              CO   
17823                 NaN      LITTLETON          DOUGLAS              CO   
17824                 NaN            NaN              NaN             NaN   

      Principal Zip         ...          Fundraising Expenses Total Expenses  \
17820         80118         ...                             0         794825   
17821         80124         ...                           150        70033.9   
17822         80435         ...                             0         754358   
17823         80125         ...                             0         260614   
17824           NaN         ...                           NaN            NaN   

      Fundraising Ratio Fundraising Expenses to Contributions  \
17820                                             0.0000        
17821                                             0.1193        
17822                                             0.0000        
17823                                             0.0000        
17824                                                NaN        

      Program Service Ratio Program Services Expenses Total Expenses  \
17820                                             0.0000               
17821                                             0.8320               
17822                                             1.0000               
17823                                             0.7492               
17824                                                NaN               

      Total Asset Amount Total Liabilities Amount  Net Assets  \
17820         17747688.0                      0.0  17747688.0   
17821            97083.0                   1682.0     95401.0   
17822            74220.0                  23462.0     50758.0   
17823            97010.0                      0.0     97010.0   
17824                NaN                      NaN         NaN   

               NTEE Code 1          NTEE Code 2          NTEE Code 3  
17820            D-ANIMALS          B-EDUCATION  O-YOUTH DEVELOPMENT  
17821  N-RECREATION,SPORTS                  NaN                  NaN  
17822          B-EDUCATION  O-YOUTH DEVELOPMENT                  NaN  
17823            D-ANIMALS          B-EDUCATION     P-HUMAN SERVICES  
17824                  NaN                  NaN                  NaN  

[5 rows x 36 columns]

In [6]:
# drop the last row with comments and null values
charities_all=charities_all[:-1]


Overall, there  were 17824 charitable organizations that submitted registration paperwork to the Colorado Secretary of State. Let's see how many of them are maintaining registration (are in good standing) by excluding those who withdrew their applications or received any special notices from the State. 


In [7]:
# Select only charitable organizations that are current on their registration
charities_current=charities_all.loc[charities_all['Current Status']=='GOOD']

In [8]:
charities_current.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12536 entries, 0 to 17823
Data columns (total 36 columns):
Principal Name of Organization                                    12536 non-null object
DBAs                                                              4233 non-null object
Registration Number                                               12536 non-null float64
EIN                                                               12536 non-null object
Principal Address 1                                               12536 non-null object
Principal Address 2                                               1552 non-null object
Principal City                                                    12536 non-null object
Principal County                                                  12478 non-null object
Principal State                                                   12531 non-null object
Principal Zip                                                     12536 non-null object
Phone       


So, there are 12536 charities that can be active in the state. Let's see how many of them are local meaning 'Principal State' == 'CO' or were originally established in Colorado ('State Established' == 'CO')


In [9]:
# Find  local charities 
charities_local= charities_current.loc[(charities_all['Principal State']=='CO') | (charities_all['State Established']=='CO')]
charities_local.reset_index(inplace=True, drop=True) #reset index in the new df

In [10]:
#Check results
charities_local.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7808 entries, 0 to 7807
Data columns (total 36 columns):
Principal Name of Organization                                    7808 non-null object
DBAs                                                              2612 non-null object
Registration Number                                               7808 non-null float64
EIN                                                               7808 non-null object
Principal Address 1                                               7808 non-null object
Principal Address 2                                               800 non-null object
Principal City                                                    7808 non-null object
Principal County                                                  7806 non-null object
Principal State                                                   7806 non-null object
Principal Zip                                                     7808 non-null object
Phone                  

In [11]:
#Find organizations that were originally established in Colorado, but moved out of state
charities_current.loc[(charities_all['Principal State'] !='CO') & (charities_all['State Established'] =='CO')]

Principal Name of Organization  \
54                                        4 D MINISTRIES   
195                        ACCENTCARE HOSPICE FOUNDATION   
338                        AFP SOUTHERN COLORADO CHAPTER   
438                                   ALEXA'S HUGS, INC.   
492          ALLIANCE FOR CONTRACEPTION IN CATS AND DOGS   
833                          AMERICAN KRATOM ASSOCIATION   
903                      AMERICAN RECORDER SOCIETY, INC.   
1009                         AMTGARD INTERNATIONAL, INC.   
1229                        ARTS COUNCIL OF MONGOLIA- US   
1281                                    ASIAN HOPE, INC.   
1389   ASSOCIATION OF CHRISTIAN MINISTRIES TO INTERNA...   
1479                      AUTHENTIC INTIMACY CORPORATION   
1637                       BEAUTY AMIDST THE ASHES, INC.   
1710                            BEST YEAR YET FOUNDATION   
1795                           BIGGER PICTURE MINISTRIES   
1852                                BLESSING RANCH, INC.   
1986                   BOULDER INSTITUTE OF MICROFINANCE   
2615            CENTER FOR AMERICAN PALEOLITHIC RESEARCH   
2738   CERTIFIED FINANCIAL PLANNER BOARD OF STANDARDS...   
2932                                 CHILDREN COME FIRST   
2963     CHILDREN'S CANCER THERAPY DEVELOPMENT INSTITUTE   
3507                           COLORADO COONHOUND RESCUE   
4210   CONFERENCE OF RESEARCH WORKERS IN ANIMAL DISEASES   
4432                    CRIME PREVENTION RESEARCH CENTER   
4564                          DANCER'S LEGACY FOUNDATION   
4884                            DEVELOPMENT IN GARDENING   
4918                             DIGNITY FREEDOM NETWORK   
5163                            DUTTON FAMILY FOUNDATION   
5549                         EQUINE EDUCATION FOUNDATION   
5606                                      ETHIOPIA READS   
...                                                  ...   
13457                                    SIGHT FOR SOULS   
13463                               SIGMA CHI FOUNDATION   
14060                               STONE GATE RESOURCES   
14255                   SUSTAINABLE TRAVEL INTERNATIONAL   
14615                          THE ASPEN INSTITUTE, INC.   
14935                     THE DELTA SIGMA PHI FOUNDATION   
14977                      THE E.P.I.C. FOUNDATION, INC.   
15120                             THE GANGAJI FOUNDATION   
15144                          THE GIFT OF LOVE MINISTRY   
15158  THE GLOBAL FUND TO FIGHT AIDS, TUBERCULOSIS AN...   
15241                   THE HISTIO CURE FOUNDATION, INC.   
15373                   THE JUSTICE MANAGEMENT INSTITUTE   
15378            THE KAPPA DELTA FOUNDATION INCORPORATED   
15601  THE NATIONAL FILM PRESERVE, LTD DBA TELLURIDE ...   
15683                          THE ORANGUTAN CONSERVANCY   
15936                        THE STURGE-WEBER FOUNDATION   
16091                          THE WILDERNESS LAND TRUST   
16263                  TOUCHING HEARTS THROUGH ATHLETICS   
16311                  TREA SENIOR CITIZENS LEAGUE, INC.   
16594                  UNITED STATES SKI TEAM FOUNDATION   
16751        URGENT ACTION FUND FOR WOMEN'S HUMAN RIGHTS   
16758                           US SNOWBOARD RACING TEAM   
16761                                      USA BADMINTON   
16774                                        USA REINING   
16809                                 UW SPORTS MINISTRY   
17136                                WE RAISE FOUNDATION   
17229                        WESTERN LANDOWNERS ALLIANCE   
17667                                        YAHERO KIDS   
17782                   YUCATAN PENINSULA MISSIONS, INC.   
17800  ZENITH OUTREACH AND ENTERPRISE NONPROFIT CORPO...   

                                                    DBAs  Registration Number  \
54                                         4D MINISTRIES         2.012302e+10   
195    ACCENTCARE HOME HEALTH OF CALIFORNIA, INC. ( D...         2.009301e+10   
338    AFP SOUTHERN COLORADO, AFPSOCO, ASSOCIATION OF...      


There are 7808 charities that either were established in Colorado or have Colorado listed as the principal state. This number includes 120 organizations that were originally established in Colorado, but have other states listed as their principal state.  For the purpose of our analysis, we will consider both types of organizations "local" to distinguish them from the country-wide and international charities operating in Colorado. 


In [12]:
#charities that have prinicipal location in Colorado by county
#charities_local.loc[charities_local['Principal State' == 'CO']].groupby(charities_local['Principal County'])

charities_local['EIN'].loc[charities_all['Principal State']=='CO'].groupby(charities_local['Principal County']).count()

Principal County
ADAMS                 186
ALAMEDA                 2
ALAMOSA                24
ARAPAHOE              461
ARCHULETA              28
ARLINGTON               1
BACA                    2
BEAUFORT                1
BENT                    2
BOULDER               479
BROOMFIELD             32
CASS                    1
CHAFFEE                23
CITY OF ALEXANDRIA      1
CLARK                   1
CLAY                    1
CLEAR CREEK            11
CO                     31
CO - COLORADO           2
COLORADO               28
COLORADO (CO)           3
COLORADO [CO]           1
CONEJOS                 3
CONNECTICUT             1
CONTRA COSTA            2
COOK                    1
COOK COUNTY             1
COSTILLA                6
CROWLEY                 1
CUSTER                 16
                     ... 
RAMSEY                  1
RIO BLANCO              5
RIO GRANDE             12
ROUTT                  62
SAGUACHE               18
SAN DIEGO               1
SAN JUAN             